projects/optic/src/utils/spec-loaders.ts

```diff
@@ -2,26 +2,146 @@ import fs from 'node:fs/promises';
 import path from 'path';
 import { promisify } from 'util';
 import { exec as callbackExec } from 'child_process';
-import { defaultEmptySpec } from '@useoptic/openapi-utilities';
-
+import { defaultEmptySpec, OpenAPIV3 } from '@useoptic/openapi-utilities';
 import {
-  ParseResult,
-  parseSpecVersion,
-  specFromInputToResults,
-} from '@useoptic/optic-ci/build/cli/commands/utils';
-import { validateOpenApiV3Document } from '@useoptic/openapi-io';
-import { filePathToGitPath } from '@useoptic/openapi-io';
+  validateOpenApiV3Document,
+  filePathToGitPath,
+  JsonSchemaSourcemap,
+  parseOpenAPIFromRepoWithSourcemap,
+  ParseOpenAPIResult,
+  parseOpenAPIWithSourcemap,
+} from '@useoptic/openapi-io';
+import { OpticCliConfig, VCS } from '../config';
+import { resolveGitRef } from './git-utils';
 
 const exec = promisify(callbackExec);
 
-export type { ParseResult };
+export type ParseResult = ParseOpenAPIResult & {
```

Comment:
I moved this code in from `optic-ci`, we're diverging the implementations since we need to also track which git sha these files came from (for tagging / uploading)